cetlvast/suites/unittest/test_pf17_string_view.cpp

```diff
@@ -0,0 +1,323 @@
+/// @file
```

Comment:
It's critical that we start with type-parameterized tests that compare the CETL implementation to the ones in our representative standard libraries like https://github.com/OpenCyphal/CETL/blob/main/cetlvast/suites/unittest/test_pf20_span.cpp does for span.

Reply:
1. See below my response, namely:
>My thinking was that it will add more work while we most probably will use char specialization only.

Could you please elaborate why it is critical to invest into what we most probably (plz correct me if I'm wrong)  will no use? Are we making CETL universal for the whole world? Honestly, it's quite exhausting to "fight" against c++14 - I believe libcyphal were long time done already if it was not support of 14 standard. And Nunavut's c++ code generation goals will be much simpler and more clean. I do understand that anyway there will be always a "race" against the latest standard and what we "have to support", so polyfills will probably still live in some form... it just concerns me (a lot!) that a lot of simple things become not so simple, and hardly contribute to the final product we deliver to customers. Sorry, for my frustration and being emotional ;)

2. BTW, you said 
>... compare the CETL implementation to the ones in our representative standard libraries ..."

I didn't quite catch what you meant by "compare"... I don't see there any comparison of what `cetl::pf20::span` and `std::span` does... if this is what you meant.

Reply:
The simple answer is that CETL's polyfill types are re-implementations from the standard and the standard defines basic_stringview then provides aliases. We can tolerate sparse implementations but must minimize deviant implementations.

As for the reasoning for C++14? It just goes to the state of the world and compatibility. The need for C++14 polyfill types will naturally diminish over time just as all software gets outdated and deprecated as the years roll by.